mail_tracking/models/mail_message.py

```diff
@@ -48,18 +47,17 @@ def tracking_status(self):
                 partner_trackings.append((
                     status, tracking.id, recipient, tracking.partner_id.id))
                 if tracking.partner_id:
-                    partners_already |= tracking.partner_id
+                    partners_already.append(tracking.partner_id.id)
             # Search all recipients for this message
-            if message.partner_ids:
-                partners |= message.partner_ids
-            if message.needaction_partner_ids:
-                partners |= message.needaction_partner_ids
-            # Remove recipients already included
-            partners -= partners_already
-            for partner in partners:
-                # If there is partners not included, then status is 'unknown'
-                partner_trackings.append((
-                    'unknown', False, partner.name, partner.id))
+            for p in message.partner_ids.filtered(
+                    lambda x: x.id not in partners_already):
+                partners[p.id] = p.name
+            for p in message.needaction_partner_ids.filtered(
+                    lambda x: x.id not in partners_already):
+                partners[p.id] = p.name
```

Comment:
If I'm not wrong, with the change in this method you're doing the same thing as before, but avoiding the use of recordsets. However recordsets are not harmful; they help making code more understandable.

Would you mind leaving it as it was? Thanks :blush: